pkg/certificate/certificate.go

```diff
@@ -90,9 +91,12 @@ func GenerateSignedCertificate(
 		return nil, nil, errors.New("while generating certificate: server name is empty")
 	}
 
+	SHA256serverNameEncrypted := sha256.New().Sum([]byte(serverName))
+
 	//nolint:exhaustruct
 	template := &x509.Certificate{
-		SerialNumber: &big.Int{}, // mandatory, but useless as it should be a unique id given to the certificate by the CA
+		//nolint:lll
+		SerialNumber: new(big.Int).SetBytes(SHA256serverNameEncrypted), // in order to have an unique ID for each website Name, we hash the website name
```

Comment:
After the analysis, I wondering if this is really mandatory. Can you try, now that you have manually added the CA root certificate to you firefox, to use the previous code to see if it works or not ?

Reply:
not working whitout the modifs : Code d’erreur : SEC_ERROR_REUSED_ISSUER_AND_SERIAL

Reply:
ok so we need to merge this PR. Thx !